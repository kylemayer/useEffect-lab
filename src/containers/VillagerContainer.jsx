import React, { useState, useEffect } from 'react';
import VillagerList from '../components/VillagerList';
import { fetchVillagers } from '../services/animalCrossingApi';

const VillagerContainer = () => {
  const [loading, setLoading] = useState(true);
  const [villagers, setVillagers] = useState([]);

  useEffect(() => {

    fetchVillagers()
      .then((villagers) => setVillagers(villagers))
      .finally(() => setLoading(false));
  }, []);

  if(loading) return <h1>Loading villagers...</h1>;

  return <VillagerList villagers={villagers} />;
};

export default VillagerContainer;
