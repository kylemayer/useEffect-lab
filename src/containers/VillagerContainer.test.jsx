import React from 'react';
import { render, screen } from '@testing-library/react';
import VillagerContainer from './VillagerContainer';
import { MemoryRouter } from 'react-router-dom';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import listVillagers from '../data/listVillagers.json';

const server = setupServer(
  rest.get('https://ac-vill.herokuapp.com/villagers', (req, res, ctx) => {
    return res(ctx.json(listVillagers));
  })
);

describe('VillagerContainer', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('displays a list of villagers on the page', async () => {
    const { container } = render(
      <MemoryRouter>
        <VillagerContainer />
      </MemoryRouter>
    );

    screen.getByText('Loading villagers...');

    const ul = await screen.findByRole('list', { name: 'villagers' });
    expect(ul).not.toBeEmptyDOMElement();
    expect(container).toMatchSnapshot();
  });
});
