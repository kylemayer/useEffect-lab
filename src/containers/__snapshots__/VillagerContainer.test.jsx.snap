// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VillagerContainer displays a list of villagers on the page 1`] = `
<div>
  <ul
    aria-label="villagers"
  >
    <li>
      <a
        href="/5f5fb4bbbfd05c2aed82e460"
      >
        <figure>
          <img
            alt="Admiral"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/ed/Admiral_NH.png/revision/latest?cb=20200802081138"
          />
          <figcaption>
            Admiral
             
            male
             - 
            aye aye
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/5f5fb4bbbfd05c2aed82e461"
      >
        <figure>
          <img
            alt="Agent S"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/a/a7/Agent_S_NH.png/revision/latest?cb=20200719092526"
          />
          <figcaption>
            Agent S
             
            female
             - 
            sidekick
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/5f5fb4bbbfd05c2aed82e462"
      >
        <figure>
          <img
            alt="Agnes"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/4/42/Agnes_NH.png/revision/latest?cb=20200804230432"
          />
          <figcaption>
            Agnes
             
            female
             - 
            snuffle
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/5f5fb4bbbfd05c2aed82e463"
      >
        <figure>
          <img
            alt="Al"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/c/c4/Al_NH.png/revision/latest?cb=20200802150354"
          />
          <figcaption>
            Al
             
            male
             - 
            hoo hoo ha CFayyyeee NH
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/5f5fb4bbbfd05c2aed82e464"
      >
        <figure>
          <img
            alt="Alfonso"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/e4/Alfonso_NH.png/revision/latest?cb=20200817164717"
          />
          <figcaption>
            Alfonso
             
            male
             - 
            it'sa me
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/5f5fb4bbbfd05c2aed82e465"
      >
        <figure>
          <img
            alt="Alice"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/e1/Alice_NH.png/revision/latest?cb=20200803181400"
          />
          <figcaption>
            Alice
             
            female
             - 
            guvnor
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/5f5fb4bbbfd05c2aed82e46c"
      >
        <figure>
          <img
            alt="Ankha"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/5/56/Ankha_NH.png/revision/latest?cb=20200817172044"
          />
          <figcaption>
            Ankha
             
            female
             - 
            me meow
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/5f5fb4bbbfd05c2aed82e466"
      >
        <figure>
          <img
            alt="Alli"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/8/8e/Alli_NH.png/revision/latest?cb=20200802061731"
          />
          <figcaption>
            Alli
             
            female
             - 
            graaagh
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/5f5fb4bbbfd05c2aed82e46b"
      >
        <figure>
          <img
            alt="Anicotti"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/8/80/Anicotti_NH.png/revision/latest?cb=20200802164003"
          />
          <figcaption>
            Anicotti
             
            female
             - 
            cannoli
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/5f5fb4bbbfd05c2aed82e46a"
      >
        <figure>
          <img
            alt="Angus"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/5/5f/Angus_NH.png/revision/latest?cb=20200802083758"
          />
          <figcaption>
            Angus
             
            male
             - 
            macmoo
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/5f5fb4bbbfd05c2aed82e467"
      >
        <figure>
          <img
            alt="Amelia"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/a/ab/Amelia_NH.png/revision/latest?cb=20200817171545"
          />
          <figcaption>
            Amelia
             
            female
             - 
            "eaglet" (GCN)"cuz"(WW & CF & NL)
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/5f5fb4bbbfd05c2aed82e46d"
      >
        <figure>
          <img
            alt="Annalisa"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/f/f0/Annalisa_NH.png/revision/latest?cb=20200802063308"
          />
          <figcaption>
            Annalisa
             
            female
             - 
            gumdrop
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/5f5fb4bbbfd05c2aed82e46f"
      >
        <figure>
          <img
            alt="Antonio"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/d4/Antonio_NH.png/revision/latest?cb=20200802063352"
          />
          <figcaption>
            Antonio
             
            male
             - 
            honk
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/5f5fb4bbbfd05c2aed82e46e"
      >
        <figure>
          <img
            alt="Annalise"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/c/c4/Annalise_NH.png/revision/latest?cb=20200802153943"
          />
          <figcaption>
            Annalise
             
            female
             - 
            nipper (NL, NH)whatever (AF+, AC, AFe+)
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/5f5fb4bbbfd05c2aed82e468"
      >
        <figure>
          <img
            alt="Anabelle"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/d8/Anabelle_NH.png/revision/latest?cb=20200802063200"
          />
          <figcaption>
            Anabelle
             
            female
             - 
            snorty
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/5f5fb4bbbfd05c2aed82e470"
      >
        <figure>
          <img
            alt="Apollo"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/2/28/Apollo_NH.png/revision/latest?cb=20200719092927"
          />
          <figcaption>
            Apollo
             
            male
             - 
            pah
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/5f5fb4bbbfd05c2aed82e469"
      >
        <figure>
          <img
            alt="Anchovy"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/ea/Anchovy_NH.png/revision/latest?cb=20200802081231"
          />
          <figcaption>
            Anchovy
             
            male
             - 
            chuurp
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/5f5fb4bbbfd05c2aed82e471"
      >
        <figure>
          <img
            alt="Apple"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/5/57/Apple_NH.png/revision/latest?cb=20200802151508"
          />
          <figcaption>
            Apple
             
            female
             - 
            cheekers
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/5f5fb4bbbfd05c2aed82e472"
      >
        <figure>
          <img
            alt="Astrid"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/d5/Astrid_NH.png/revision/latest?cb=20200802154954"
          />
          <figcaption>
            Astrid
             
            female
             - 
            my pet
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/5f5fb4bbbfd05c2aed82e473"
      >
        <figure>
          <img
            alt="Audie"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/1/1b/Audie_NH.png/revision/latest?cb=20200829012735"
          />
          <figcaption>
            Audie
             
            female
             - 
            foxtrot
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/5f5fb4bbbfd05c2aed82e474"
      >
        <figure>
          <img
            alt="Aurora"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/5/59/Aurora_NH.png/revision/latest?cb=20200202215217"
          />
          <figcaption>
            Aurora
             
            female
             - 
            b-b-baby
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/5f5fb4bbbfd05c2aed82e475"
      >
        <figure>
          <img
            alt="Ava"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/c/cc/Ava_NH.png/revision/latest?cb=20200609055741"
          />
          <figcaption>
            Ava
             
            female
             - 
            beaker
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/5f5fb4bbbfd05c2aed82e476"
      >
        <figure>
          <img
            alt="Avery"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/dd/Avery_NH.png/revision/latest?cb=20200608222354"
          />
          <figcaption>
            Avery
             
            male
             - 
            skree-haw
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/5f5fb4bbbfd05c2aed82e478"
      >
        <figure>
          <img
            alt="Baabara"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/df/Baabara_NH.png/revision/latest?cb=20200802165552"
          />
          <figcaption>
            Baabara
             
            female
             - 
            daahling
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/5f5fb4bbbfd05c2aed82e47b"
      >
        <figure>
          <img
            alt="Barold"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/9/9d/Barold_NH.png/revision/latest?cb=20200802141140"
          />
          <figcaption>
            Barold
             
            male
             - 
            cubby
          </figcaption>
        </figure>
      </a>
    </li>
  </ul>
</div>
`;
